src/olympia/addons/serializers.py

```diff
@@ -109,10 +116,26 @@ def get_version_license_url(self, obj):
         # given License. However, since we're serializing through a nested
         # serializer, we cheat and use `instance.version_instance` which is
         # set by SimpleVersionSerializer.to_representation() while serializing.
-        if hasattr(obj, 'version_instance'):
+        # Only get the version license url for non-builtin licenses.
+        if not obj.builtin and hasattr(obj, 'version_instance'):
             return absolutify(obj.version_instance.license_url())
         return None
 
+    def get_name(self, obj):
+        # See if there is a license constant
+        license_constant = obj._constant
+        if not license_constant:
+            # If not fall back on the name in the database.
+            return self.db_name.get_attribute(obj)
+        else:
+            request = self.context.get('request', None)
+            if request and request.method == 'GET' and 'lang' in request.GET:
+                # A single lang requested so return a flat string
+                return unicode(license_constant.name)
+            else:
+                # Otherwise mock the dict, and it's going to be lang default.
+                return {settings.LANGUAGE_CODE: unicode(license_constant.name)}
```

Comment:
This is all a nitpick because we are not supposed to need to call this API without a `lang` yet, but:

Technically, the middleware uses `Accept-Language` as well, so the current lang could be different than `settings.LANGUAGE_CODE`. Use `request.LANG` instead, unless there is somehow no `request`, in which case falling back to `settings.LANGUAGE_CODE`.